Add pytest fixture support

belgeu opened the pull request:
Currently test has no setup/teardown workflow.

Add pytest's builtin fixture support in order to provide ability to
execute [setup] and [teardown] steps before and after each test case run.

belgeu committed:
Add pytest fixture support

Currently test has no setup/teardown workflow.

Add pytest's builtin fixture support in order to provide ability to
execute [setup] and [teardown] steps before and after each test case run.

extsoft commented on python/webdriver_test.py:
```diff
@@ -27,20 +27,25 @@
-    b = webdriver.Firefox()
+@pytest.fixture
+def setup(request):
+    request.b = webdriver.Firefox()
```
Don't add driver to requests.  Use 
```
driver = webdriver.Firefox()
.....
return driver
```

extsoft commented on python/webdriver_test.py:
```diff
@@ -27,20 +27,25 @@
-def test_01():
-    b = webdriver.Firefox()
+@pytest.fixture
+def setup(request):
```
Pls rename fixture (`browser` etc.)

belgeu replied:
Changed fixture name to 'browser'.

extsoft commented on python/webdriver_test.py:
```diff
@@ -27,20 +27,25 @@
-    b = webdriver.Firefox()
+@pytest.fixture
+def setup(request):
+    request.b = webdriver.Firefox()
```
Don't add driver to requests.  Use 
```
driver = webdriver.Firefox()
.....
return driver
```

belgeu replied:
Modified fixture to return webdriver object separately from request.

belgeu committed:
Address PR comments for b0d8151 commit

- change fixture name to 'browser'
- initialize webdriver separately from request

extsoft commented on python/webdriver_test.py:
```diff
@@ -27,20 +27,25 @@
+    return driver
+
+def test_01(browser):
+    b = browser
```
Why do you create this `b`?

belgeu replied:
The idea is to keep code a bit shorter, but i see your point. Switched webdriver naming to "browser", it certainly improves code readability. Thanks.

belgeu committed:
Remove browser reference to variable

Improves code readability.
